Session/Home/HomeViewModel.swift

```diff
@@ -301,23 +301,33 @@ public class HomeViewModel {
     
     // MARK: - Functions
     
-    public func delete(threadId: String, threadVariant: SessionThread.Variant) {
+    public func delete(threadId: String, threadVariant: SessionThread.Variant, force: Bool = false) {
+        
+        func delete(_ db: Database, threadId: String) throws {
+            _ = try SessionThread
+                .filter(id: threadId)
+                .deleteAll(db)
+        }
+        
         Storage.shared.writeAsync { db in
             switch threadVariant {
                 case .closedGroup:
-                    try MessageSender.leave(
-                        db,
-                        groupPublicKey: threadId,
-                        deleteThread: true
-                    )
+                    if force {
+                        try delete(db, threadId: threadId)
+                    } else {
+                        try MessageSender.leave(
+                            db,
+                            groupPublicKey: threadId,
+                            deleteThread: true
+                        )
+                    }
                     
                 case .openGroup:
                     OpenGroupManager.shared.delete(db, openGroupId: threadId)
-                    _ = try SessionThread
-                        .filter(id: threadId)
-                        .deleteAll(db)
+                    try delete(db, threadId: threadId)
                     
-                default: break
+                default:
+                    try delete(db, threadId: threadId)
```

Comment:
This can probably be cleaned up a little:
```
switch (threadVariant, force) {
    case (.closedGroup, false):
        try MessageSender.leave(
            db,
            groupPublicKey: threadId,
            deleteThread: true
        )

    case (.openGroup, _):
        OpenGroupManager.shared.delete(db, openGroupId: threadId)

    default:
        try delete(db, threadId: threadId)
}
```